File: apps/cargo-tracker/app/cargotracker/routing/domain/model/repositories/VoyageRepository.scala
package cargotracker.routing.domain.model.repositories

import cargotracker.routing.domain.model.aggregates.Voyage
import cargotracker.routing.domain.model.valueobjects.VoyageNumber

/** 航海集約の永続化ポート。 */
trait VoyageRepository:
  def findByVoyageNumber(voyageNumber: VoyageNumber): Option[Voyage]
  def findAll(): Seq[Voyage]
  def save(voyage: Voyage): Unit

File: apps/cargo-tracker/src/contexts/routing/domain/repository/voyage-repository.ts
import type { Voyage } from '../model/voyage.js';

/** 航海リポジトリ出力ポート */
export interface VoyageRepository {
  save(voyage: Voyage): Promise<number>;
  findByVoyageNumber(voyageNumber: string): Promise<Voyage | null>;
  update(voyage: Voyage): Promise<void>;
}
